src/services/equipmentService.js

```diff
@@ -0,0 +1,84 @@
+class EquipmentService {
+  constructor(conn) {
+    this.conn = conn;
+    this.changed = [];
+    this.added = [];
+    this.removed = [];
+    this.result = {
+      changed: this.changed,
+      added: this.added,
+      removed: this.removed,
+    };
+  }
+
+  equipmentById(id) {
+    const query = 'SELECT * FROM equipment WHERE id = ?;';
+    return new Promise((resolve, reject) => {
+      this.conn.query(query, [id], (err, row) => {
+        if (err) {
+          reject(new Error(500));
+        } else {
+          row.length > 0 ? resolve(row[0]) : reject(new Error(400));
+        }
+      });
+    });
+  }
+
+  checkIfActiveExists(type, heroId, itemId) {
+    const query = 'SELECT * FROM equipment WHERE heroId = ? AND type = ? AND active = 1;';
+
+    return new Promise((resolve, reject) => {
+      this.conn.query(query, [heroId, type], (err, row) => {
+        err ? reject(new Error(500)) : resolve(row[0]);
+      });
+    }).then((res) => {
+      if (res) {
+        if (res.id !== itemId) {
+          this.changeActive(0, res.id);
+          res.active = 0;
+          delete res.heroId;
+          return res;
+        }
+        return new Error(400);
+      }
+      return new Error(418);
+    });
+  }
+
+  changeActive(changeTo, id) {
+    const query = 'UPDATE equipment SET active = ? WHERE id = ?;';
+
+    this.conn.query(query, [changeTo, id], (err) => {
+      if (err) Promise.reject(new Error(500));
+    });
+  }
+
+  async equip(item) {
+    const equipmentToChange = await this.equipmentById(item.id);
+
+    if (equipmentToChange instanceof Object && equipmentToChange.id) {
+      const unequipPreviouslyEquipped = await this.checkIfActiveExists(equipmentToChange.type, equipmentToChange.heroId, item.id);
+
+      if (!(unequipPreviouslyEquipped instanceof Error)) this.changed.push(unequipPreviouslyEquipped);
+      if (!(this.changeActive(1, item.id) instanceof Error)) {
+        equipmentToChange.active = 1;
+        delete equipmentToChange.heroId;
+        this.changed.push(equipmentToChange);
+      }
+    }
+  }
+
+  async use(item) {
+    this.changed.length = 0;
+    this.added.length = 0;
+    this.removed.length = 0;

```

Comment:
if you are always resetting the fields, you might as well create a new object each time.
i understand that you wanted to access these variables from any point of the code, but this is not a good practice. in this scope these variables act as globals and using globals can go wrong especially when used heavily. let's talk about this during mentortime.

change the code structure so that you dont store information in the class for a request.